plugin/kubernetai/kubernetai_test.go

```diff
@@ -125,8 +167,26 @@ func TestKubernetai_AutoPath(t *testing.T) {
 					},
 				},
 			},
-			want: []string{"test-1.svc.cluster.local.", "svc.cluster.local.", "cluster.local.", "test-1.svc.fluster.local.", "svc.fluster.local.", "fluster.local.", ""},
-			ip:   "172.17.0.7",
+			want:     []string{"test-1.svc.cluster.local.", "svc.cluster.local.", "cluster.local.", "test-1.svc.fluster.local.", "svc.fluster.local.", "fluster.local.", ""},
+			ip:       "172.17.0.7",
+			podCache: true,
+		},
+		{
+			name:   "standard autopath lookup fluster in cluster.local with podMode = podModeDisabled",
+			fields: defaultK8iConfig,
+			args: args{
+				state: request.Request{
+					W: w,
+					Req: &dns.Msg{
+						Question: []dns.Question{
+							{Name: "svc-d.test-2.svc.fluster.local.svc.cluster.local.", Qtype: 1, Qclass: 1},
+						},
+					},
+				},
+			},
+			want:     []string{"svc.cluster.local.", "cluster.local.", "svc.fluster.local.", "fluster.local.", ""},
```

Comment:
why the `""`?

Reply:
That's needed in the autopath.
https://github.com/coredns/coredns/blob/master/plugin/autopath/autopath.go#L23-L24

Basicaly the autopath will go through the searchList / searchPath
And append it to what it consider to be the base query
for example
you request `test.toto` in your pod 
the resolv.conf search path add `.svc.cluster.local`
coredns receive `test.toto.svc.cluster.local`

The autopath will remove `svc.cluster.local` [code for this here](https://github.com/coredns/coredns/blob/b455f86824a1c2108b305589190e964b5528fed3/plugin/autopath/autopath.go#L95)
and then try to append each element of the searchPath sequentialy until it find an answer for it.
`test.toto` + `.svc.cluster.local`
`test.toto` + `.cluster.local`
`test.toto` + `` .  <-- The empty string is used, so that it tries to resolve what coredns consider to be the raw request.